### PR TITLE
Handle setting TextEditingController text to null

### LINKED ISSUE #34847: TextEditingController ignores text = null
## Steps to Reproduce

```
  final editName = TextEditingController();
...
    editName.text = 'qwe';
    editName.text = null;
```

Expected: empty text field.
Observed: text 'qwe'.

I see this behavior as counter intuitive. I would suggest adding the` ?? '' `check into `TextEditingController ::   set text(String newText)`.

Workaround: 
`    editName.text = map['field'] ?? '';`

### PR DESCRIPTION
**Pull request by justinmc**
## Description

Setting `textEditingController.text = null` previously didn't clear the text, but with this PR it does.

## Related Issues

Closes https://github.com/flutter/flutter/issues/34847

## Tests

I tested clearing the field via:

  * Setting the controller value to empty string.
  * TextFormField reset
  * TextFormField didChange

## Breaking Change

This will break anyone that is currently setting TextEditingController's text value to null.  To migrate, they should likely should set it to empty string.

### CODE REVIEW

**Comment by goderbauer** [comment]
(PR triage): @justinmc Do you still have plans for this one?

**Comment by justinmc** [comment]
Yes, though probably taking a different approach per https://github.com/flutter/flutter/issues/34847#issuecomment-713960970.  I'll investigate later today and decide to close or not.

File: packages/flutter/test/widgets/editable_text_test.dart [hunk 1]
```diff
@@ -5391,77 +5391,114 @@ void main() {
     expect(focusNode.hasFocus, false);
   });
 
-  testWidgets('TextEditingController.clear() behavior test', (WidgetTester tester) async {
-    // Regression test for https://github.com/flutter/flutter/issues/66316
-    final List<MethodCall> log = <MethodCall>[];
-    SystemChannels.textInput.setMockMethodCallHandler((MethodCall methodCall) async {
-      log.add(methodCall);
+  group('TextEditingController', () {
```
Comment (justinmc) [inline packages/flutter/test/widgets/editable_text_test.dart:5394]: This diff is confusing, but what I did was add the test on the next line and group it with the following existing test.

**Comment by justinmc** [comment]
@cbracken Ready for review.  I need to check if this breaks any customers, and if so I'll go through the full breaking change process.

**Comment by justinmc** [comment]
I believe this is not breaking internal tests or customer tests.

**Review by cbracken — APPROVED** [review]
LGTM stamp from a Japanese personal seal

**Pull request merged**